examples/semi_supervised/plot_semi_supervised_newsgroups.py

```diff
@@ -3,17 +3,42 @@
 Semi-supervised Classification on a Text Dataset
 ================================================
 
-In this example, semi-supervised classifiers are trained on the 20 newsgroups
-dataset (which will be automatically downloaded).
+This example demonstrates the effectiveness of semi-supervised learning
+in text classification when labeled data is scarce.
+We compare four different approaches:
 
-You can adjust the number of categories by giving their names to the dataset
-loader or setting them to `None` to get all 20 of them.
+1. Supervised learning using 100% of labeled data (baseline)
```

Comment:
In wouldn't call this a baseline: if we do not have access to 100% labeled data, it will probably not be possible to reach the performance of this model. I would rather call it a "best case scenario".